app/controllers/admin/schedules_controller.rb

```diff
@@ -4,72 +4,37 @@ class SchedulesController < Admin::BaseController
     # the schedule of a conference, which should not be accessed in the first place
     load_and_authorize_resource :conference, find_by: :short_title
     load_and_authorize_resource :program, through: :conference, singleton: true
+    load_and_authorize_resource :schedule, through: :program
+    load_resource :event_schedules, through: :schedule
+    load_resource :selected_schedule, through: :program, singleton: true
     load_resource :venue, through: :conference, singleton: true
 
-    skip_before_filter :verify_authenticity_token, only: [:update]
-    layout 'schedule'
+    def index; end
 
-    def show
-      authorize! :update, @program.events.new
-      if @conference.nil?
-        redirect_to admin_conference_index_path
-        return
-      end
-      @dates = @conference.start_date..@conference.end_date
-      if @venue && @venue.rooms.any?
-        @rooms = @venue.rooms
+    def create
+      if @schedule.save
+        redirect_to action: 'show', id: @schedule.id
       else
-        @rooms = [ Room.new(name: 'No Rooms!', size: 0) ]
-      end
-    end
-
-    def update
-      authorize! :update, @program.events.new
-      event = Event.where(guid: params[:event]).first
-      error_message = nil
-      if event.nil?
-        error_message = "Could not find event GUID: #{params[:event]}"
-      end
-
-      if params[:date] == 'none'
-        event.start_time = nil
-        event.room = nil
-        event.save!
-        render json: { 'status' => 'ok' }
-        return
-      end
-      room = Room.where(guid: room_params).first
-      if room.nil?
-        error_message = "Could not find room GUID: #{params[:room]}"
+        redirect_to admin_conference_schedules_path(conference_id: @conference.short_title),
+                    error: 'Could not create schedule'
```

Comment:
It is shown

![image](https://cloud.githubusercontent.com/assets/16052290/17448708/17f5bcce-5b56-11e6-9692-1fa87a4e015d.png)


Unless you mean something else... :confused: